edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Docker.Test/RuntimeInfoProviderTest.cs

```diff
@@ -332,5 +332,91 @@ public async Task GetModuleLogsTest()
             receivedLogs = Encoding.UTF8.GetString(buffer, 0, readBytes);
             Assert.Equal(dummyLogs, receivedLogs);
         }
+
+        [Fact]
+        [Unit]
+        public async Task GetModules_Handles_NotFoundException()
+        {
+            var missingContainerName = "container2";
+            var containersToReturn = new[] { "container1", "container2", "container3" };
+
+            var dockerClient = SetupDockerClient(missingContainerName, containersToReturn);
+            var provider = await RuntimeInfoProvider.CreateAsync(dockerClient);
+
+            var moduleInfo = await provider.GetModules(CancellationToken.None);
+
+            var actualContainerNames = moduleInfo
+                                            .Select(m => m.Name);
+
+            var expectedContainerNames = containersToReturn
+                                            .Where(c => !c.Equals(missingContainerName))
+                                            .Select(c => c.Substring(1))    // cuts the first letter
+                                            .Concat(new[] { "dgeAgent" });  // added anyway
+
+            Assert.Equal(expectedContainerNames, actualContainerNames);
+        }
+
+        [Fact]
+        [Unit]
+        public async Task GetModules_Handles_MissingEdgeAgent()
+        {
+            var missingContainerName = "edgeAgent";
+            var containersToReturn = new[] { "container1", "container2", "container3" };
+
+            var dockerClient = SetupDockerClient(missingContainerName, containersToReturn);
+            var provider = await RuntimeInfoProvider.CreateAsync(dockerClient);
+
+            var moduleInfo = await provider.GetModules(CancellationToken.None);
+
+            var actualContainerNames = moduleInfo
+                                            .Select(m => m.Name);
+
+            var expectedContainerNames = containersToReturn
+                                            .Select(c => c.Substring(1));    // cuts the first letter
+
+            Assert.Equal(expectedContainerNames, actualContainerNames);
+        }
+
+        private IDockerClient SetupDockerClient(string nonExistingContainer, params string[] containers)
+        {
+            var dockerClient = new Mock<IDockerClient>();
+            var dockerContainer = new Mock<IContainerOperations>();
+            var dockerSystem = new Mock<ISystemOperations>();
+            var containerList = containers.Select(c => new ContainerListResponse { ID = c }).ToList();
```

Comment:
```suggestion
            var containerList = containers.Select(c => new ContainerListResponse { ID = $"/{s}" }).ToList();
```

Reply:
adding the trailing '/' a few lines below